bin/p2-rctl/main.go

```diff
@@ -0,0 +1,208 @@
+	CMD_DISABLE  = "disable"
+)
+
+var (
+	consulUrl   = kingpin.Flag("consul", "The hostname and port of a consul agent in the p2 cluster. Defaults to 0.0.0.0:8500.").String()
+	consulToken = kingpin.Flag("token", "The consul ACL token to use. Empty by default.").String()
+	headers     = kingpin.Flag("header", "An HTTP header to add to requests, in KEY=VALUE form. Can be specified multiple times.").StringMap()
+	https       = kingpin.Flag("https", "Use HTTPS").Bool()
+
+	cmdCreate       = kingpin.Command(CMD_CREATE, "Create a new replication controller")
+	createManifest  = cmdCreate.Flag("manifest", "manifest file to use for this replication controller").Short('m').Required().String()
+	createNodeSel   = cmdCreate.Flag("node-selector", "node selector that this replication controller should target").Short('n').Required().String()
+	createPodLabels = cmdCreate.Flag("pod-label", "a pod label, in LABEL=VALUE form, to add to this replication controller. Can be specified multiple times.").Short('p').StringMap()
+
+	cmdDelete   = kingpin.Command(CMD_DELETE, "Delete a replication controller")
+	deleteID    = cmdDelete.Arg("id", "replication controller uuid to delete").Required().String()
+	deleteForce = cmdDelete.Flag("force", "delete even if desired replicas > 0").Short('f').Bool()
+
+	cmdReplicas = kingpin.Command(CMD_REPLICAS, "Set desired replica count of a replication controller")
+	replicasID  = cmdReplicas.Arg("id", "replication controller uuid to modify").Required().String()
+	replicasNum = cmdReplicas.Arg("replicas", "number of replicas desired").Required().Int()
+
+	cmdList  = kingpin.Command(CMD_LIST, "List replication controllers")
+	listJSON = cmdList.Flag("json", "output the entire JSON object of each replication controller").Short('j').Bool()
+
+	cmdGet      = kingpin.Command(CMD_GET, "Get replication controller")
+	getID       = cmdGet.Arg("id", "replication controller uuid to get").Required().String()
+	getManifest = cmdGet.Flag("manifest", "print just the manifest of the replication controller").Short('m').Bool()
+
+	cmdEnable = kingpin.Command(CMD_ENABLE, "Enable replication controller")
+	enableID  = cmdEnable.Arg("id", "replication controller uuid to enable").Required().String()
+
+	cmdDisable = kingpin.Command(CMD_DISABLE, "Disable replication controller")
+	disableID  = cmdDisable.Arg("id", "replication controller uuid to disable").Required().String()
+)
+
+func main() {
+	kingpin.Version(version.VERSION)
+	cmd := kingpin.Parse()
+
+	logger := logging.NewLogger(logrus.Fields{})
+	logger.Logger.Formatter = &logrus.TextFormatter{}
+	opts := kp.Options{
+		Address: *consulUrl,
+		Token:   *consulToken,
+		Client:  net.NewHeaderClient(*headers, http.DefaultTransport),
+		HTTPS:   *https,
+	}
+	client := kp.NewConsulClient(opts)
+	rcStore := rcstore.NewConsul(client, 3, logger)
+	rctl := RCtl{
+		rcs:    rcStore,
+		logger: logger,
+	}
+
+	switch cmd {
+	case CMD_CREATE:
+		rctl.Create(*createManifest, *createNodeSel, *createPodLabels)
```

Comment:
Since we have multiple subcommands we can't use Kingpin's `.Required()` directive for flags. Need parameter validation here.